ui/src/datasource/DatasourceCreate.vue

```diff
@@ -67,6 +67,10 @@ export default class DatasourceCreate extends Vue {
       type: 'JSON',
       parameters: {}
     },
+    dataSchema: {
```

Comment:
I think we will eventually simplify to `dataSchema: <schema>`, that's the only thing the datasource config should need. Converting the data with a mode into the schema should happen independently on the ui, where it comes from shouldn't matter to the final config. Mode and the actual data will then only exist in the config step ui component